pkgs/shells/fish/default.nix

```diff
@@ -142,8 +138,6 @@ let
             "$out/share/fish/functions/prompt_pwd.fish"
       sed -i "s|nroff |${groff}/bin/nroff |"               \
             "$out/share/fish/functions/__fish_print_help.fish"
-      sed -i "s|/sbin /usr/sbin||" \
-            "$out/share/fish/functions/__fish_complete_subcommand_root.fish"
```

Comment:
It looks like `__fish_complete_subcommand_root.fish` got effectively inlined into `share/completions/doas.fish` and `share/completions/sudo.fish`, so instead of deleting this, we should retain it but run it on those two files instead.

Though looking further, those files also gained an inclusion of `/usr/local/sbin` as well. Or more specifically, both files have the following line:

```fish
set -lx -a PATH /usr/local/sbin /sbin /usr/sbin
```

I admit I'm not clear as to why these completions are modifying the PATH, or why we're then un-modifying it.

Reply:
The commit that originally introduced that behavior was https://github.com/NixOS/nixpkgs/commit/5cffa00443e222c6c56c346de96734b08f2fa060,
which seems to have been working around hardcoded binaries. Nowadays, it appears
that these places use `command` to test the existence of the binary before
executing (though I didn't check every file). I did two things to test whether
or not this find-replace of binaries is still necessary (not foolproof, but
satisfied my curiosity):

* spun up a NixOS VM and overlayed `fish` to have an empty `postInstall`
* ran `fish` inside of `nix-shell --pure` with an empty `postInstall` and a
temporary `HOME` (`nix-shell -p fish --pure` and then `sh -c 'env HOME=$(mktemp
-d) fish'`

and noticed that in neither scenario did I observe `fish` complain about
anything. I ran the following commands from the root of nixpkgs without any
issue:

* `type _`
* `_ asdf`
* `ls`
* `ls | grep pkgs`

We would probably still want the `etcConfigAppendixText` and `fishPreInitHooks`
to be written out in `postInstall`, but everything else appears unnecessary.

I could remove all of the dead `sed` calls and add whatever new ones are
necessary, or I could wipe out `postInstall` except for `etcConfigAppendixText`
and `fishPreInitHooks`.

cc @ocharles -- as the maintainer, what would you prefer?

Reply:
This particular sed in question is mucking with the completions for `sudo` and `doas`. I wish I knew why Fish thought it had to muck with the `PATH` in general for completions of those two commands.

Reply:
That is strange. Especially since overwriting with the older `sudo` completion without the `PATH` fiddling, everything works as expected on my system (same amount of suggested binaries). I'll follow your suggestion and `sed` those as well, and open a follow-up PR addressing the necessity of `postInstall`, so we can get this merged sooner rather than later.

Something to be done at a later date is ask upstream whether or not this `PATH` fiddling is really necessary.